packages/block-editor/src/components/border-radius-control/index.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	RangeControl,
+	__experimentalParseUnit as parseUnit,
+	__experimentalUseCustomUnits as useCustomUnits,
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import AllInputControl from './all-input-control';
+import InputControls from './input-controls';
+import LinkedButton from './linked-button';
+import {
+	getAllValue,
+	getAllUnit,
+	hasDefinedValues,
+	hasMixedValues,
+} from './utils';
+
+const DEFAULT_VALUES = {
+	topLeft: null,
+	topRight: null,
+	bottomLeft: null,
+	bottomRight: null,
+};
+const MIN_BORDER_RADIUS_VALUE = 0;
+const MAX_BORDER_RADIUS_VALUES = {
+	px: 100,
+	em: 20,
+	rem: 20,
+};
+
+/**
+ * Control to display border radius options.
+ *
+ * @param {Object}   props          Component props.
+ * @param {Function} props.onChange Callback to handle onChange.
+ * @param {Object}   props.values   Border radius values.
+ *
+ * @return {WPElement}              Custom border radius control.
+ */
+export default function BorderRadiusControl( { onChange, values } ) {
+	const [ isLinked, setIsLinked ] = useState(
+		! hasDefinedValues( values ) || ! hasMixedValues( values )
+	);
+
+	const units = useCustomUnits( { availableUnits: [ 'px', 'em', 'rem' ] } );
```

Comment:
Does this need to use the `spacing.units` or `layout.units` available via `theme.json`? I've noticed it's one of the few places where these aren't used so are not configurable by the theme.

Reply:
Good point.

I can see theme developers wishing to allow different units for borders to those of spacing or layouts as well. A PR allowing configuration of border units can be found in https://github.com/WordPress/gutenberg/pull/33315.